src/BluetoothDeviceDriver/BluetoothDeviceDriver.ino

```diff
@@ -139,10 +132,7 @@ void loop() {
   if (recording) {
     uint32_t available_samples = (recording_idx + BUFFER_SIZE - read_idx) % BUFFER_SIZE;
     if (available_samples >= CHUNK_SIZE) {
-      Serial.print("Sending ");
-      Serial.print(available_samples);
-      Serial.println(" samples");
-
+      Serial.println("Sending " + String(available_samples) + " samples");
```

Comment:
Probably better to get rid of it. We've read writing to serial while sending BLE might cause issues. Would be better if you can put this behind a debug flag with a #define

Reply:
Addressed your comment. Thank you for pointing that out!